src/main/java/frc/robot/RobotContainer.java

```diff
@@ -105,7 +116,30 @@ private void configureDefaultCommands() {
                         () -> true));
     }
 
-    private void configureButtonBindings() {}
+    private Command updateScoreState() {
+        return (currentState.calculateTargets().andThen(currentState.prepareSubsystems()))
+                .repeatedly();
+    }
+
+    private void configureButtonBindings() {
+        xboxController
+                .a()
+                .onTrue(
+                        Commands.runOnce(() -> currentState = shootState)
+                                .andThen(updateScoreState()));
+        xboxController
+                .b()
+                .onTrue(
+                        Commands.runOnce(() -> currentState = ampState)
+                                .andThen(updateScoreState()));
+        xboxController
+                .x()
+                .onTrue(
+                        Commands.runOnce(() -> currentState = climbState)
+                                .andThen(updateScoreState()));
+
+        xboxController.rightTrigger(0.1).onTrue(new ProxyCommand(() -> currentState.score()));
```

Comment:
```suggestion
        xboxController.rightTrigger(0.1).onTrue(Commands.deferredProxy(() -> currentState.score()));
```
(The line would have been shorter without the directly-importing-`ProxyCommand` cheat 😉)
However, a deferred command fits here better (proxies should be used to fork-off a command group). Did you choose to go with a proxy command to omit the requirements set?

Edit: apparently a deferred proxy will be deprecated soon so it should be changed (https://github.com/wpilibsuite/allwpilib/pull/6324)

Reply:
It doesn't matter that it is going deprecated because we won't be using this code outside the season probably. Resolve please